src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -48,6 +60,33 @@ public void add(Person toAdd) {
         internalList.add(toAdd);
     }
 
+    /**
+     * Gets the type of a person in the list.
+     */
+    public Class<? extends Person> getPersonType(Person person) {
+        requireNonNull(person);
+        if (person instanceof Guest) {
+            return Guest.class;
+        } else {
+            assert(person instanceof Vendor);
```

Comment:
Good use of assertion here to make sure the program fails if it's not either a Guest or a Vendor!